contracts/mediator/HomeMediator.sol

```diff
@@ -0,0 +1,86 @@
+pragma solidity 0.4.24;
+
+import "./BasicMediator.sol";
+import "../interfaces/IForeignMediator.sol";
+import "../interfaces/ISimpleBridgeKitty.sol";
+
+contract HomeMediator is BasicMediator {
+    function passMessage(address _from, uint256 _tokenId, bytes _metadata) internal {
+        bytes4 methodSelector = IForeignMediator(0).handleBridgedTokens.selector;
+        bytes memory data = abi.encodeWithSelector(methodSelector, _from, _tokenId);
+
+        bytes32 dataHash = keccak256(data);
+        setMessageHashTokenId(dataHash, _tokenId);
+        setMessageHashRecipient(dataHash, _from);
+        setMessageHashMetadata(dataHash, _metadata);
+
+        bridgeContract().requireToPassMessage(mediatorContractOnOtherSide(), data, requestGasLimit());
+    }
+
+    function handleBridgedTokens(address _recipient, uint256 _tokenId, bytes _metadata) external {
+        require(msg.sender == address(bridgeContract()));
+        require(bridgeContract().messageSender() == mediatorContractOnOtherSide());
+
+        mintToken(_recipient, _tokenId, _metadata);
+    }
+
+    function mintToken(address _recipient, uint256 _tokenId, bytes _metadata) internal {
+        ISimpleBridgeKitty(erc721token()).mint(
+            _tokenId,
+            getMetadataBoolValue(_metadata, 2), // isReady
+            getMetadataUintValue(_metadata, 3), // cooldownIndex
+            getMetadataUintValue(_metadata, 4), // nextActionAt
+            getMetadataUintValue(_metadata, 5), // siringWithId
+            getMetadataUintValue(_metadata, 6), // birthTime
+            getMetadataUintValue(_metadata, 7), // matronId
+            getMetadataUintValue(_metadata, 8), // sireId
+            getMetadataUintValue(_metadata, 9), // generation
+            getMetadataUintValue(_metadata, 10), // genes
+            _recipient
+        );
+    }
+
+    function bridgeSpecificActionsOnTokenTransfer(address _from, uint256 _tokenId) internal {
+        bytes memory metadata = getMetadata(_tokenId);
```

Comment:
In the ForeignMediator implementation `getMetadata` is part of `passMessage` method. Why is it different here?

Reply:
It is different because here it burns the token before calling `passMessage` so it need to get the metadata first. An option could be to burn the token after calling `passMessage` so we can get the metadata as part of  `passMessage`

Reply:
thanks! it is OK for me.